pulp_container/app/tasks/recursive_remove.py

```diff
@@ -1,15 +1,28 @@
 from django.db.models import Q
-from pulpcore.plugin.models import Content
+from pulpcore.plugin.models import Content, Repository
 
 from pulp_container.app.models import (
     Blob,
-    ContainerRepository,
     Manifest,
     MEDIA_TYPE,
     Tag,
 )
 
 
+def recursive_remove_content_find_tags(repository_pk, manifest_pk):
```

Comment:
we need to add blobs here as well. And manifests in case what you are trying to remove by digest happens to be a manifest_list. Now the trick is to remove only those blobs that are not referenced by any other image within that repo; remove only those image manifests that are not referenced in other manifest lists within that repo and those that are not tagged. All this logic is respected in the task `recursive_remove_content`.
I suggest using it, just in the `content_unit` already provide pks of the manifest/manifest-list and pks of all tags you will find referenced by that object.

Reply:
you can find all the pks i was talking about in the viewet and pass it to the task  from the viewset.